# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1533664482 -7200
#      Tue Aug 07 19:54:42 2018 +0200
# Node ID 3c9134d90d6ed5d46a970d656bf304b0d30edf3d
# Parent  785eef952919d925316a8f8baf561b1f3008b326
Prevent immediate updates of GUI workspace view when using clear in a script (bug #54443).

* variables.cc (Fclear): Remove explicit call to octave_link::set_workspace.

diff --git a/libinterp/corefcn/variables.cc b/libinterp/corefcn/variables.cc
--- a/libinterp/corefcn/variables.cc
+++ b/libinterp/corefcn/variables.cc
@@ -2339,18 +2339,16 @@ without the dash as well.
                   octave_class::clear_exemplar_map ();
                   symtab.clear_all ();
                 }
               else
                 {
                   do_clear_symbols (symtab, argv, argc, idx, exclusive);
                 }
             }
-
-          octave_link::set_workspace ();
         }
     }
 
   return ovl ();
 }
 
 DEFUN (whos_line_format, args, nargout,
        doc: /* -*- texinfo -*-
